providers/fips/self_test_data.inc

```diff
@@ -824,188 +824,190 @@ static const ST_KAT_DRBG st_kat_drbg_tests[] =
 
 #ifndef OPENSSL_NO_DH
 /* DH KAT */
+/* RFC7919 FFDHE2048 p */
```

Comment:
Why is this change required?
What's wrong with the existing self test?

Reply:
[NIST SP 800-56Ar3](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-56Ar3.pdf) says in section 5.5.1.1 on DH params:

> Two classes of domain parameters are approved for FFC key agreement: a class of “safe” domain parameters that are associated with approved safe-prime groups, and a class of “FIPS 186-type” domain parameters that conform to one of the FIPS 186-type parameter-size sets that are listed in Table 1.

A footnote on the FIPS 186-type domain parameters points to "FIPS 186, Digital Signature Standard (DSS)". [FIPS 186-4](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-4.pdf) does specify those domain params in section 4 "DSA" subsection 4.3.1 "Domain Parameter Generation" and Appendix A. However, the newer version [FIPS 186-5](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-5.pdf) no longer specifies DSA at all; it just says:

> Prior versions of this standard specified the DSA. This standard no longer approves the DSA for digital signature generation. However, the DSA may be used to verify signatures generated prior to the implementation date of this standard. See FIPS 186-4 for the specifications for the DSA.

You don't need generation of FIPS 186-type domain parameters for signature verification.

We've been told by our lab that once the transition period for DSA ends, submissions will no longer be able to claim DSA, and with that FIPS 186-type domain parameters will automatically become non-approved as well. I believe the transition period has not yet started, but when it does, use of a FIPS 186-type group in the self-test would become unapproved after a year. For this reason, we swapped it with a safe-prime group and suggest you do the same to be future-proof.

Reply:
Thanks for the explanation.